### PR TITLE
Heartbeats and Securities for Redis dying

### PR DESCRIPTION
**Pull request by exception**
This Pull request adds:
- Heartbeats to servers to detect if they die.
- Prevention of redis dying, if it does, we reconnect.

### CODE REVIEW

**Comment by mkotb** [comment]
Is this tested?

**Comment by exception** [comment]
Deploying this to a working network to test would be a good idea, i assumed you had a test setup locally. @mkotb

File: daemon-bash/src/main/java/io/minecloud/daemon/MineCloudDaemon.java [hunk 1]
```diff
@@ -227,6 +237,20 @@ private MineCloudDaemon(Properties properties) {
                 }
             });
 
+            nodeServers.forEach(server ->  {
+                try (Jedis jedis = this.redis.grabResource()) {
```
Comment (mkotb) [inline daemon-bash/src/main/java/io/minecloud/daemon/MineCloudDaemon.java]: Do you have to grab a new resource for _every_ server? Seems a bit excessive

**Comment by exception** [comment]
@mkotb Fixed in the latest commit.

**Comment by mkotb** [comment]
please test

**Comment by mkotb** [comment]
thanks for this I actually needed it, turns out

**Pull request merged**